src/main/java/com/fasterxml/jackson/databind/DeserializationContext.java

```diff
@@ -465,28 +465,18 @@ public final boolean hasSomeOfFeatures(int featureMask) {
     public final JsonParser getParser() { return _parser; }
 
     public final Object findInjectableValue(Object valueId,
-            BeanProperty forProperty, Object beanInstance)
+            BeanProperty forProperty, Object beanInstance, Boolean optional)
         throws JsonMappingException
     {
         if (_injectableValues == null) {
-            final JacksonInject.Value injectableValue = getAnnotationIntrospector()
-                    .findInjectableValue(forProperty.getMember());
-            // Default optional-or-not from global setting:
-            boolean optional = !isEnabled(DeserializationFeature.FAIL_ON_UNKNOWN_INJECT_VALUE);
-            // but may be overridden on per-property basis:
-            if (injectableValue != null) {
-                // `null` means "use defaults" (global setting)
-                if (injectableValue.getOptional() != null) {
-                    optional = injectableValue.getOptional();
-                }
-            }
-            if (optional) {
+            if (!isEnabled(DeserializationFeature.FAIL_ON_UNKNOWN_INJECT_VALUE)
```

Comment:
and here change so that annotation has precedence (DeserializationFeature only used if `optional` not `null`).